src/ssl.c

```diff
@@ -32598,10 +32599,13 @@ WOLFSSL_EC_KEY *wolfSSL_d2i_ECPrivateKey(WOLFSSL_EC_KEY **key, const unsigned ch
         return NULL;
     }
 
-    if (wc_ecc_import_private_key(*in, (word32)len, NULL, 0,
-            (ecc_key*)eckey->internal) != MP_OKAY) {
-        WOLFSSL_MSG("wc_ecc_import_private_key error");
-        goto error;
+    if (wc_EccPrivateKeyDecode(*in, &idx, (ecc_key*)eckey->internal,
+            (word32)len) != 0) {
+        if (wc_ecc_import_private_key(*in, (word32)len, NULL, 0,
+                (ecc_key*)eckey->internal) != MP_OKAY) {
+            WOLFSSL_MSG("wc_ecc_import_private_key error");
+            goto error;
+        }
```

Comment:
This should just be an error. We shouldn't attempt to read the raw key.

Reply:
Fixed.
Changed i2d to encode into DER.